### PR TITLE
Option to show what nation you're joining when you are invited to join a town.

### PR DESCRIPTION
**Pull request by AlessioGr**
#### Description: 

This closes following sugestion: https://github.com/TownyAdvanced/Towny/issues/5524 (it explains it very well)
Old:
image
New:
image

____
#### New Nodes/Commands/ConfigOptions: 

New Node: `INVITE_SYSTEM_INVITATION_SHOW_NATION_NAME` / `invite_system.invitation_show_nation_name`
New Translation: `you_have_been_invited_to_join3: 'You have been invited to join &b%s&9 (Nation: &3%s&9)'`

____
#### Relevant Towny Issue ticket:

https://github.com/TownyAdvanced/Towny/issues/5524

____
- [X] I have tested this pull request for defects on a server. 

By making this pull request, I represent that I have the right to waive copyright and related rights to my contribution, and agree that all copyright and related rights in my contributions are waived, and I acknowledge that the TownyAdvanced organization has the copyright to use and modify my contribution under the Towny [License](https://github.com/LlmDl/Towny/blob/master/LICENSE.md) for perpetuity.

### CODE REVIEW

File: src/com/palmergames/bukkit/towny/TownyMessaging.java [hunk 1]
```diff
@@ -352,24 +352,29 @@ public static void sendTitleMessageToNation(Nation nation, String title, String
...
+		if (invite.getSender() instanceof Town town) { // Town invited Resident
+			String firstLine;
+			if(TownySettings.isInvitationShowNationName() && town.getNationOrNull() != null){
+				firstLine = translator.of("invitation_prefix") + translator.of("you_have_been_invited_to_join3", senderName, town.getNationOrNull().getName());
+			}else{
+				firstLine = translator.of("invitation_prefix") + translator.of("you_have_been_invited_to_join2", senderName);
+			}
+			String confirmLine = TownySettings.getAcceptCommand() + " " + senderName;
+			String cancelLine = TownySettings.getDenyCommand() + " " + senderName;
+			sendInvitationMessage(player, firstLine, confirmLine, cancelLine);
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/TownyMessaging.java:364]:
```suggestion
			String firstline = town.hasNation()
			        ? translator.of("invitation_prefix") + translator.of("you_have_been_invited_to_join3", senderName, town.getNationOrNull())
			        : translator.of("invitation_prefix") + translator.of("you_have_been_invited_to_join2", senderName);
			String confirmline = TownySettings.getAcceptCommand() + " " + senderName;
			String cancelline = TownySettings.getDenyCommand() + " " + senderName;
			sendInvitationMessage(player, firstline, confirmline, cancelline);

```

File: src/com/palmergames/bukkit/towny/TownyMessaging.java [hunk 2]
```diff
@@ -352,24 +352,29 @@ public static void sendTitleMessageToNation(Nation nation, String title, String
...
 			}
 			if (invite.getReceiver() instanceof Nation) { // Nation allied Nation
-				String firstline = translator.of("invitation_prefix") + translator.of("you_have_been_requested_to_ally2", senderName);
-				String confirmline = "n ally accept " + senderName;
-				String cancelline = "n ally deny " + senderName;
-				sendInvitationMessage(player, firstline, confirmline, cancelline);
+				String firstLine = translator.of("invitation_prefix") + translator.of("you_have_been_requested_to_ally2", senderName);
+				String confirmLine = "n ally accept " + senderName;
+				String cancelLine = "n ally deny " + senderName;
+				sendInvitationMessage(player, firstLine, confirmLine, cancelLine);
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/TownyMessaging.java:377]: These can go back to their original casing.

File: src/com/palmergames/bukkit/config/ConfigNodes.java [hunk 3]
```diff
@@ -1648,6 +1648,11 @@
 			"# +------------------------------------------------------+ #",
 			"############################################################",
 			""),
+	INVITE_SYSTEM_INVITATION_SHOW_NATION_NAME(
+		"invite_system.invitation_show_nation_name",
+		"true",
+		"",
+		"# If you invite another player to your town, this will decide if it should show the nation name together with the town name, or just the town name."),
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/config/ConfigNodes.java:1655]: This doesn't really need to be a config option, I think it's a good quality-of-life improvement on its own.

File: src/com/palmergames/bukkit/towny/TownySettings.java [hunk 4]
```diff
@@ -2332,7 +2332,10 @@ public static boolean getDefaultPermission(TownBlockOwner owner, PermLevel level
 			case OUTSIDER -> getDefaultOutsiderPermission(owner, type);
 		};
 	}
-
+	public static boolean isInvitationShowNationName() {
+		return getBoolean(ConfigNodes.INVITE_SYSTEM_INVITATION_SHOW_NATION_NAME);
+	}
+
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/TownySettings.java:2338]: Not required.

**Review by LlmDl — CHANGES_REQUESTED** [review]

File: src/com/palmergames/bukkit/config/ConfigNodes.java (hunk 3 above)
Comment (AlessioGr) [reply src/com/palmergames/bukkit/config/ConfigNodes.java]: Removed

**Review by AlessioGr — COMMENTED** [review]
Removed config options

File: src/com/palmergames/bukkit/towny/TownyMessaging.java [hunk 5]
```diff
@@ -352,11 +352,14 @@ public static void sendTitleMessageToNation(Nation nation, String title, String
...
-		if (invite.getSender() instanceof Town) { // Town invited Resident
-			String firstline = translator.of("invitation_prefix") + translator.of("you_have_been_invited_to_join2", senderName);
+		if (invite.getSender() instanceof Town town) { // Town invited Resident
+			String firstline = town.hasNation()
+			        ? translator.of("invitation_prefix") + translator.of("you_have_been_invited_to_join3", senderName, town.getNationOrNull())
+			        : translator.of("invitation_prefix") + translator.of("you_have_been_invited_to_join2", senderName);
 			String confirmline = TownySettings.getAcceptCommand() + " " + senderName;
 			String cancelline = TownySettings.getDenyCommand() + " " + senderName;
 			sendInvitationMessage(player, firstline, confirmline, cancelline);
+
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/TownyMessaging.java:362]: Remove diff

File: src/com/palmergames/bukkit/towny/TownySettings.java [hunk 6]
```diff
@@ -2332,7 +2332,7 @@ public static boolean getDefaultPermission(TownBlockOwner owner, PermLevel level
 			case OUTSIDER -> getDefaultOutsiderPermission(owner, type);
 		};
 	}
-
+
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/TownySettings.java:2335]: Remove diff

**Review by LlmDl — CHANGES_REQUESTED** [review]
Just a couple of minor whitespacing changes and I think this one is ready.

**Review by AlessioGr — COMMENTED** [review]
Fixed

**Pull request merged**